src/coreComponents/physicsSolvers/multiphysics/poromechanicsKernels/ThermalMultiphasePoromechanics_impl.hpp

```diff
@@ -201,27 +199,11 @@ computeBodyForce( localIndex const k,
   {
     GEOS_UNUSED_VAR( mixtureDensity );
 
-    arraySlice1d< real64 const, constitutive::multifluid::USD_PHASE - 2 > const phaseMassDensity = m_fluidPhaseMassDensity[k][q];
-    arraySlice2d< real64 const, constitutive::multifluid::USD_PHASE_DC - 2 > const dPhaseMassDensity = m_dFluidPhaseMassDensity[k][q];
-    arraySlice1d< real64 const, compflow::USD_PHASE - 1 > const phaseVolFrac = m_fluidPhaseVolFrac[k];
-    arraySlice2d< real64 const, compflow::USD_PHASE_DC - 1 > const dPhaseVolFrac = m_dFluidPhaseVolFrac[k];
-
-    // Step 1: compute fluid total mass density and its derivatives
-
-    real64 dTotalMassDensity_dTemperature = 0.0;
-
-    for( integer ip = 0; ip < m_numPhases; ++ip )
-    {
-      dTotalMassDensity_dTemperature += dPhaseVolFrac( ip, Deriv::dT ) * phaseMassDensity( ip )
```

Comment:
I suppose this means that these terms cancel out and we should end up with a value that is numerically close to zero?

Reply:
If we highlight in the body force term the dependencies on the primary variables we have:

$$\rho \mathbf{g}=(1-\phi(\epsilon_v(\mathbf{u}),p,T))\rho_{\text{solid}}(p,T)\mathbf{g}+\phi(\epsilon_v(\mathbf{u}),p,T)\sum_{c}\rho_c\mathbf{g}$$

We have a direct dependency on temperature exclusively on porosity at the moment -- eventually on the solid phase density $\rho_s$ (not considered at the moment in the code). I agree that one can probably show that looping over phases produces  `dTotalMassDensity_dTemperature = 0`.